### ISSUE TITLE
Rule: Semicolon spacing

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#629]
This rule should warn when there is a space before a semicolon, such as:

``` js
var foo = "bar" ;
```

It should not warn when the semicolon is the non-whitespace first character on a line, such as:

``` js
!(function(){}());
```

The rule ID should be `no-space-before-semi` and be on by default.
